File: raspberryPi3/bcm2835/python/blink.py
'''
Program: blink.py
Description: Blinking led
			using bcm2835 library
Author: Aldo Nunez
'''
# Import GPIO module
import RPi.GPIO as GPIO
import time 

def main ():
	# port number
	PORT = 7;
	# time delay in milliseconds
	T = 0.05 	# 0.05 ~ 50 ms
	
	GPIO.setup ( PORT, GPIO.OUT )
	while True:
		GPIO.output ( PORT, HIGH )
		time.sleep ( T )
		GPIO.output ( PORT, LOW ) 
		time.sleep ( T )

if __name__  == "__main__":
	main ()

File: raspberryPi3/bcm2835/c++/src/blink.cpp
//
// Program: blink.cpp
// Description: Blink led using 
// 				bcm2835 library
// Author: Aldo Nunez
//
//
#include <iostream>
#include <bcm2835.h>

using namespace std;

int
main ()
{
	// J8_07 ~ Pin 7. Check bcm2835.h, line 842 
	const int PORT = RPI_BPLUS_GPIO_J8_07;

	// delay in milliseconds 
	const int T = 50;

	// setup 
    if ( !bcm2835_init () )
	{
		cerr <<  "Failing to setup port" << endl ;
		return 1;
	}

    // set the port as output 
    bcm2835_gpio_fsel ( PORT, BCM2835_GPIO_FSEL_OUTP );
	
	cout << "blinking..." << endl;

	while ( 1  )
    {
        // Turn on 
        bcm2835_gpio_set ( PORT );
        // wait 
        bcm2835_delay ( T );
        
        // turn off
        bcm2835_gpio_clr ( PORT );
	    //  wait 
	    bcm2835_delay ( T );
    }

    bcm2835_close ();
    return 0;
 }

